### PR TITLE
Release pipeline fail on Azure Git repo with LFS 

### PR DESCRIPTION
**Pull request by Roman-Shchukin**
**Description**:
Release pipeline fail on Azure Git repo with LFS 

**Steps to reproduce**:
1. add a png filter to git attributes of an existing repo without prior lfs filters *.png filter=lfs diff=lfs merge=lfs -text
2. add a png file to the repo using regular 'git add', 'git commit' commands
3. git push
4. try to pull azure repo git artifact in a release pipeline
Result: checkout fail

**Solution**:
Added setting `GIT_LFS_SKIP_SMUDGE=1` to disable LFS fetch/checkout when the customer didn't enable LFS in their pipeline definition

**Related issue:** https://github.com/microsoft/build-task-team/issues/3662

### CODE REVIEW

File: src/Agent.Worker/Build/GitSourceProvider.cs [hunk 1]
```diff
@@ -363,8 +363,19 @@ public abstract class GitSourceProvider : SourceProvider, ISourceProvider
...
+            if (!gitLfsSupport)
+            {
+                gitEnv["GIT_LFS_SKIP_SMUDGE"] = "1";
+                executionContext.Debug("GIT_LFS_SKIP_SMUDGE variable set to 1");
+            }
+
             // Initialize git command manager
             _gitCommandManager = HostContext.GetService<IGitCommandManager>();
+            await _gitCommandManager.LoadGitExecutionInfo(executionContext, useBuiltInGit: !preferGitFromPath, gitEnv);
             await _gitCommandManager.LoadGitExecutionInfo(executionContext, useBuiltInGit: !preferGitFromPath);
```
Comment (KonstantinTyukalov) [inline src/Agent.Worker/Build/GitSourceProvider.cs:379]: Are we need to call LoadGitExecutionInfo second time?

Comment (Roman-Shchukin) [reply src/Agent.Worker/Build/GitSourceProvider.cs]: Fixed

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by LiliaSabitova — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Review by wojgasior — APPROVED** [review]

**Pull request merged**